src/test/java/org/elasticsearch/test/engine/MockInternalEngine.java

```diff
@@ -66,22 +66,30 @@ public MockContext(Random random, boolean wrapReader, Class<? extends FilterDire
 
     private MockContext mockContext;
 
-    public MockInternalEngine(EngineConfig config) throws EngineException {
-        super(config);
-        Settings indexSettings = config.getIndexSettings();
-        final long seed = indexSettings.getAsLong(ElasticsearchIntegrationTest.SETTING_INDEX_SEED, 0l);
-        Random random = new Random(seed);
-        final double ratio = indexSettings.getAsDouble(WRAP_READER_RATIO, 0.0d); // DISABLED by default - AssertingDR is crazy slow
-        Class<? extends AssertingDirectoryReader> wrapper = indexSettings.getAsClass(READER_WRAPPER_TYPE, AssertingDirectoryReader.class);
-        boolean wrapReader = random.nextDouble() < ratio;
-        if (logger.isTraceEnabled()) {
-            logger.trace("Using [{}] for shard [{}] seed: [{}] wrapReader: [{}]", this.getClass().getName(), shardId, seed, wrapReader);
+    public MockInternalEngine(EngineConfig config, boolean skipInitialTranslogRecovery) throws EngineException {
+        super(config, skipInitialTranslogRecovery);
+    }
+
+    private synchronized MockContext getMockContext() {
```

Comment:
oh man we get the searcher in teh ctor in the uppper class that requires to call getSearcher which needs the context...